more/auction/auction.js

```diff
@@ -0,0 +1,198 @@
+      const natExtentOps = makeNatExtentOps();
+      return (
+        natExtentOps.includes(leftAssetDesc, rightAssetDesc) &&
+        !natExtentOps.equals(leftAssetDesc, rightAssetDesc)
+      );
+    }
+
+    E(timerP)
+      .delayUntil(deadline)
+      .then(() => {
+        // hold auction unless too few Bids or minPrice not met
+        if (
+          bidsReceived < minBidCount ||
+          strictlyIncludes(minPrice, secondPrice)
+        ) {
+          cancelExcept(undefined);
+          sellerRefund.res(escrowedGoods.p);
+          auctionComplete.reject(
+            bidsReceived < minBidCount
+              ? 'too few bids'
+              : 'minimum price not met.',
+          );
+          return;
+        }
+        cancelExcept(bestBidderP);
+        const bestBidAgencySeatP = agencySeatsP.get(bestBidderP);
+        const paidAssetDescP = E(bestBidAgencySeatP).consummateDeal(
+          bestPrice,
+          secondPrice,
+          escrowedGoods.p,
+        );
+        sellerWinnings.res(E(bestBidAgencySeatP).getWinnings());
+        sellerRefund.res();
+        paidAssetDescP.then(paidAssetDesc => {
+          auctionComplete.res(paidAssetDesc.extent);
+        });
+      });
+
+    function addNewBid(paymentP, buyerSeatP, agencySeatP) {
+      return E(currencyAssayP)
+        .claimAll(paymentP)
+        .then(currencyPaymentP => {
+          return E(currencyPaymentP)
+            .getBalance()
+            .then(assetDesc => {
+              const { extent } = assetDesc;
+              E(buyerSeatP).offer(currencyPaymentP, escrowTerms);
+              bidsReceived += 1;
+              if (strictlyIncludes(extent, bestPrice)) {
+                bestBidderP = buyerSeatP;
+                [bestPrice, secondPrice] = [extent, bestPrice];
+              } else if (strictlyIncludes(extent, secondPrice)) {
+                secondPrice = extent;
+              }
+              agencySeatsP.set(buyerSeatP, agencySeatP);
+              return extent;
+            });
+        });
+    }
+
+    const bidderMaker = harden({
+      // Each call on makeBidderSeat() will return a bidderSeat invite, which
+      // allows the holder to make offers and ensures that the money will be
+      // returned unless they get the goods.
+      makeBidderSeat() {
+        const seats = E(agencyEscrowInstallationP).spawn(escrowTerms);
+        const agencySeatP = seats.then(pair => pair.agency);
+        const buyerSeatP = seats.then(pair => pair.buyer);
+
+        bidderSeatCount += 1;
+        const bidderSeat = harden({
+          offer(paymentP) {
+            return addNewBid(paymentP, buyerSeatP, agencySeatP);
+          },
+          getWinnings() {
+            return E(buyerSeatP).getWinnings();
+          },
+          getRefund() {
+            return E(buyerSeatP).getRefund();
+          },
+          // Can the bidder cancel? Not currently.
+        });
+        const bidderId = `bidder${bidderSeatCount}`;
```

Comment:
I suppose "Id" is short for "identifier" so I shouldn't complain. But I only use `Id` for unforgeable things. Perhaps I should consistently use `token` or something instead?